cpu/sam0_common/periph/dac.c

```diff
@@ -110,13 +110,28 @@ int8_t dac_init(dac_t line)
 
     _dac_init_clock(line);
 
-    /* Settings can only be changed when DAC is disabled, reset config */
-    DAC->CTRLA.reg = DAC_CTRLA_SWRST;
+    /* Settings can only be changed when DAC is disabled */
+    DAC->CTRLA.reg &= ~DAC_CTRLA_ENABLE;
```

Comment:
Since there are only two bits, you could also just
```suggestion
    DAC->CTRLA.reg = 0;
```

Reply:
For some CPUs there are three bits. One of them is the bit to enable operation during standby power mode:

```
typedef union {
  struct {
    uint8_t  SWRST:1;          /*!< bit:      0  Software Reset                     */
    uint8_t  ENABLE:1;         /*!< bit:      1  Enable                             */
    uint8_t  RUNSTDBY:1;       /*!< bit:      2  Run in Standby                     */
    uint8_t  :5;               /*!< bit:  3.. 7  Reserved                           */
  } bit;                       /*!< Structure used for bit  access                  */
  uint8_t reg;                 /*!< Type      used for register access              */
} DAC_CTRLA_Type;
```

Reply:
But now you are writing the entire `CTRLA` reg anyway at the end, you might as well set it all to 0 here :wink:

Reply:
But it saves two instructions - but yea, not really that much relevant on init.